system/routes_test.go

```diff
@@ -27,22 +44,187 @@ func TestUsers(t *testing.T) {
+
+		req.AddCookie(&http.Cookie{
+			Name:     "jwt",
+			Value:    tokenString,
+			Domain:   ".localhost",
+			Expires:  time.Now().Add(time.Hour),
+			HttpOnly: true,
+			MaxAge:   50000,
+			Path:     "/auth",
+		})
+
+		recorder := httptest.NewRecorder()
+		Routes().ServeHTTP(recorder, req)
+		resp := recorder.Result()
+
+		fmt.Println(">>> (request)")
+		fmt.Println(request(req))
+		fmt.Println("----")
+		fmt.Println("<<< (response)")
+		fmt.Println(response(resp))
+
+		jr, err := decodeJson(resp.Body)
+		assert(t, err == nil, "Error decoding response body: %+v", err)
+		assert(t, jr.Error.Message == "failed to authorize request: signature is invalid", "Expected error 'failed to authorize request: signature is invalid' got: %+v", jr.Error.Message)
+	}
+
+	// Send "Login" request without parameters.
+	{
+		req, err := http.NewRequest("POST", "http://localhost/auth/login", nil)
+		assert(t, err == nil, "Error creating request: %+v", err)
+
+		recorder := httptest.NewRecorder()
+		Routes().ServeHTTP(recorder, req)
+		resp := recorder.Result()
+
+		fmt.Println(">>> (request)")
+		fmt.Println(request(req))
+		fmt.Println("----")
+		fmt.Println("<<< (response)")
+		fmt.Println(response(resp))
+
+		jr, err := decodeJson(resp.Body)
+		assert(t, err == nil, "Error decoding response body: %+v", err)
+		assert(t, jr.Error.Message == "missing form body", "Expected error 'missing form body' got: %+v", jr.Error.Message)
+	}
+
+	// Send "Login" request with missing user.
+	{
+		json := []byte(`{"username":"test123","password":"test123"}`)
+
+		req, err := http.NewRequest("POST", "http://localhost/auth/login", bytes.NewBuffer(json))
+		req.Header.Set("Content-Type", "application/json")
+
+		assert(t, err == nil, "Error creating request: %+v", err)
+
+		recorder := httptest.NewRecorder()
+		Routes().ServeHTTP(recorder, req)
+		resp := recorder.Result()
+
+		fmt.Println(">>> (request)")
+		fmt.Println(request(req))
+		fmt.Println("----")
+		fmt.Println("<<< (response)")
+		fmt.Println(response(resp))
+
+		jr, err := decodeJson(resp.Body)
+		assert(t, err == nil, "Error decoding response body: %+v", err)
+		assert(t, jr.Error.Message == "crust.auth.repository.UserNotFound", "Expected error 'crust.auth.repository.UserNotFound' got: %+v", jr.Error.Message)
+	}
+
+	// Create user.
+	user := &systemTypes.User{
+		ID:       1337,
+		Username: "johndoe",
+	}
+	{
+		err := user.GeneratePassword("johndoe123")
+		assert(t, err == nil, "Error generating password: %+v", err)
+	}
+	{
+		userAPI := systemRepository.User(context.Background(), nil)
+		_, err := userAPI.Create(user)
+		assert(t, err == nil, "Error when inserting user: %+v", err)
+	}
+
+	// Send "Login" request with existing user.
+	{
+		form := url.Values{}
+		form.Add("username", "johndoe")
+		form.Add("password", "johndoe123")
+
+		req, err := http.NewRequest("POST", "http://localhost/auth/login", strings.NewReader(form.Encode()))
+		req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
```

Comment:
I'd like you to test with a json post body here separately (`application/json` content-type).